### PR TITLE
profiler documentation

### PR DESCRIPTION
**Pull request by laura-makdah**
## Description 

CLI documentation for the gas profiler https://github.com/MystenLabs/sui/pull/15775

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docs/content/references/cli/client.mdx [hunk 1]
```diff
@@ -239,6 +239,52 @@ Execution finished successfully. Local and on-chain effects match.
...
+```shell
+speedscope <PATH/PROFILE-OUTPUT-FILE>
+```
+
+When looking at the profile in speedscope, there are 3 different views available from the top left menu: `Timer Order`,
+`Left Heavy`, and `Sandwich`. In each view, each bar's vertical width corresponds to the percentage of gas consumption incurred by the function,
+and you can hover your mouse over a bar or click a bar to see the precise amount of gas consumed in nano-sui.
+
+`Time Order` will show the callstack of function invocations from left to right in the order
+of invocation, while `Left Heavy` will combine repeated sequences of of nested invocations into a single combined call stack,
```
Comment (dariorussi) [inline docs/content/references/cli/client.mdx:273]: double "of" - "of of"

File: docs/content/references/cli/client.mdx [hunk 2]
```diff
@@ -239,6 +239,52 @@ Execution finished successfully. Local and on-chain effects match.
...
+and will display these from left to right by total incurred gas consumption per combined call stack. This is useful when there have
+been hundreds of repeated calls to the same function to quickly observe the total gas consumption over all calls to that function.
+In both these views the top section can be clicked and dragged to zoom in and out over different sections of the profile.
+
+`Sandwich` view shows a list of discrete values that correspond to gas consumption per function, with `Total` showing
+gas cost incurred in all the functions called by the function, and `Self` showing the gas cost done by only the given function.
+
+Observing which functions contain the most gas consumption provides a useful starting point for optimization of gas cost
```
Comment (dariorussi) [inline docs/content/references/cli/client.mdx:281]: I would not not talk about optimizations much at this point, I would rather have people use this to check where the gas usage is higher. That can later prompt optimizations but I would not start there... 
not sure I am making sense here...

**Review by dariorussi — APPROVED** [review]
cool thanks

File: docs/content/references/cli/client.mdx [hunk 3]
```diff
@@ -239,19 +239,67 @@ Execution finished successfully. Local and on-chain effects match.
...
+
+To install speedscope run
+
+```bash npm2yarn
+npm install -g speedscope
+```
+
+To open a profile in speedscope run
+
+```bash npm2yarn
```
Comment (ronny-mysten) [inline docs/content/references/cli/client.mdx:267]:
```suggestion
```bash
```
My bad

File: docs/content/references/cli/client.mdx [hunk 4]
```diff
@@ -239,19 +239,67 @@ Execution finished successfully. Local and on-chain effects match.
 Use `sui client replay-batch --path <FILEPATH>` to replay several transactions listed in a newline-separated file. This
 will verify that all transactions local execution results match the effects on-chain.
 
+### Profile a transaction
+
+Use the `sui client profile-transaction --tx-digest <TRANSACTION-DIGEST>` command to re-execute a transaction locally
+and produce a gas profile. Similar to the `replay` command, this command fetches the transaction dependencies from the
+Full node specified in the client environment that are needed to execute the transaction. During the local execution of the transaction,
```
Comment (stefan-mysten) [inline docs/content/references/cli/client.mdx:246]:
```suggestion
full node specified in the client environment that is used to execute the transaction. During the local execution of the transaction,
```

File: docs/content/references/cli/client.mdx [hunk 5]
```diff
@@ -239,19 +239,67 @@ Execution finished successfully. Local and on-chain effects match.
...
+of invocation, while **Left Heavy** combines repeated sequences of nested invocations into a single combined call stack. **Left Heavy** 
+displays these sequences from left to right by total incurred gas consumption per combined call stack. This is useful when there have
+been hundreds of repeated calls to the same function to quickly observe the total gas consumption over all calls to that function.
+In both these views, you can click the top section and drag to zoom in and out over different sections of the profile.
+
+**Sandwich** view shows a list of discrete values that correspond to gas consumption per function, with **Total** showing
+gas cost incurred in all the functions called by the function, and **Self** showing the gas cost done by only the given function.
+
+Observing a transaction's gas consumption provides useful insight of expected gas cost usage of a smart contract.
+When developing a smart contract, you can [run a local network](../../guides/developer/getting-started/local-network.mdx) and publish the package to the local network. Then create a transaction that calls
```
Comment (stefan-mysten) [inline docs/content/references/cli/client.mdx:285]: I think you wouldn't need the extra `../../`. 
```suggestion
When developing a smart contract, you can [run a local network](/guides/developer/getting-started/local-network.mdx) and publish the package to the local network. Then create a transaction that calls
```

File: docs/content/references/cli/client.mdx [hunk 6]
```diff
@@ -239,19 +239,67 @@ Execution finished successfully. Local and on-chain effects match.
...
 This example also makes use of `sui move` commands. To learn more about those commands, see [Sui Move CLI](./move.mdx).
 
-1. Open a terminal or console to the root of your local Sui repository and navigate to the move_tutorial example.
+1. Open a terminal or console to the root of your local Sui repository and navigate to the `move_tutorial` example.
     ```shell
     cd examples/move
     cd first_package
     ```
-1. Switch to the Devnet network. This command uses an alias, so the `devnet` value might be different for you, depending on the alias name set in your configuration.
+1. Switch to the Devnet network. This command uses an alias, so the `devnet` value might be different for you, depending on the alias name set in your configuration (use `sui client envs` for a list of your aliases).
```
Comment (stefan-mysten) [inline docs/content/references/cli/client.mdx:302]:
```suggestion
1. Switch to the Devnet network. This command uses an alias, so the `devnet` value might be different for you, depending on the alias name set in your configuration (use `sui client envs` for a list of your defined networks and their aliases).
```

**Review by stefan-mysten — APPROVED** [review]
Looks good, thanks @laura-makdah. If I may suggest, it might be useful to organize it in some sort of sections. For example, maybe something like this? Not sure how well it would fit, but breaking the text might help with a better flow.
#### Installation

#### Understanding the output

#### Common use cases

**Pull request merged**